### PR TITLE
Make BufferSearch less wide

### PR DESCRIPTION
**Pull request by ConradIrwin**
This also adds some "responsiveness" so that UI elements are hidden
before everything has to be occluded

Release Notes:

- Improved search UI. It now works in narrower panes, and avoids scrolling the editor on open.

Screenshot 2024-04-11 at 21 33 17
Screenshot 2024-04-11 at 21 33 45
Screenshot 2024-04-11 at 21 34 00
Screenshot 2024-04-11 at 21 37 03

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Zed
 
-[![CI](https://github.com/zed-industries/zed/actions/workflows/ci.yml/badge.svg)](https://github.com/zed-industries/zed/actions/workflows/ci.yml)
+[![CI](https://github.com/zed-industries/zed/actions/workflows/ci.yml/badge.svg)](https://github.com/zed-industries/ze34actions/workflows/ci.yml)
```
Comment (bennetbo) [inline README.md:3]: Was this intentional? Looks wrong to me

**Comment by bennetbo** [comment]
There are some issues with overlapping text now:
image

**Comment by ConradIrwin** [comment]
I'm not sure how to make the right button render its background underneath the numbers (the left arrow isn't quite so broken feeling), so I've gone back to the arrangement where the count is outside the arrows.

https://github.com/zed-industries/zed/pull/10474

**Comment by baldwindavid** [comment]
🥳 This is a really nice improvement! No longer jarring and much less intrusive. Thank you!

**Pull request merged**